app/javascript/Pages/SingleClass.jsx

```diff
@@ -0,0 +1,220 @@
+import React from "react";
+import axios from "axios";
+import PropTypes from "prop-types";
+import { useState } from "react";
+import { Link } from "@inertiajs/inertia-react";
+import Layout from "./Layout";
+
+const SingleClass = (props) => {
+  const [click, setClick] = useState("about");
+
+  const handleClick = (newState) => setClick(newState);
+  const avatar = (image) => {
+    return `${image}?tr=,r-max`;
+  };
+
+  let requestInfo = props.request;
+
+  const buttonOnClick = () => {
+    switch (buttonState) {
+      case "pending":
+        //cancel request
+        setButtonState("loading");
+
+        axios
+          .put(`/api/match_request/${requestInfo.match_id}/`, {
+            status: "cancelled",
+            headers: window.defaultHeaders,
+          })
+          .then((response) => {
+            console.log(response.data);
+
+            setButtonState("available");
+          })
+          .catch((error) => {
+            if (error.response.data.error) {
+              alert(error.response.data.error);
+            }
+            setButtonState("pending");
+          });
+
+        break;
+
+      case "available":
+        //send request
+        setButtonState("loading");
+
+        axios
+          .post(`/api/classes/${props.class.id}/request/`, {
+            headers: window.defaultHeaders,
+          })
+          .then((response) => {
+            console.log(response.data);
```

Comment:
Same here, i guess. You should probably search through the entire project if there any console.logs forgotten and remove them.